File: app/src/main/java/rssdemo/andras/hu/rssdemo/network/FeedApi.java
package rssdemo.andras.hu.rssdemo.network;


import com.einmalfel.earl.EarlParser;

import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;
import java.io.InputStream;
import java.net.URL;
import java.util.zip.DataFormatException;

public class FeedApi {

    public com.einmalfel.earl.Feed loadFeed(String feedUrl) throws IOException, DataFormatException, XmlPullParserException {
        InputStream inputStream = new URL(feedUrl).openConnection().getInputStream();
        return EarlParser.parseOrThrow(inputStream, 0);
    }
}

File: app/src/main/java/rssdemo/andras/hu/rssdemo/network/FeedApi.kt
package rssdemo.andras.hu.rssdemo.network


import com.einmalfel.earl.EarlParser
import org.xmlpull.v1.XmlPullParserException
import java.io.IOException
import java.net.URL
import java.util.zip.DataFormatException

class FeedApi {

    @Throws(IOException::class, DataFormatException::class, XmlPullParserException::class)
    fun loadFeed(feedUrl: String): com.einmalfel.earl.Feed {
        val inputStream = URL(feedUrl).openConnection().getInputStream()
        return EarlParser.parseOrThrow(inputStream, 0)
    }
}
